concrete-core/src/specification/engines/glwe_ciphertext_discarding_tensor_product_computation.rs

```diff
@@ -0,0 +1,63 @@
+use super::engine_error;
+use crate::prelude::GlweCiphertext32;
+use crate::specification::engines::AbstractEngine;
+use crate::specification::entities::GlweCiphertextEntity;
+
+engine_error! {
+    GlweCiphertextDiscardingTensorProductError for GlweCiphertextDiscardingTensorProductEngine @
+    PolynomialSizeMismatch => "The polynomial size of the input and output GLWE ciphertexts must be\
+     the same.",
+    InputGlweDimensionMismatch => "The GLWE dimension of the input ciphertexts must be the same.",
+    OutputGlweDimensionMismatch => "The GLWE dimension of the output ciphertext is incorrect"
+}
+
+impl<EngineError: std::error::Error> GlweCiphertextDiscardingTensorProductError<EngineError> {
+    pub fn perform_generic_checks<InputCiphertext1, InputCiphertext2, OutputCiphertext>(
+        input1: &InputCiphertext1,
+        input2: &InputCiphertext2,
+        output: &OutputCiphertext,
+    ) -> Result<(), Self>
+    where
+        InputCiphertext1: GlweCiphertextEntity,
```

Comment:
This is correct, change files to match this, we shouldn't have any types (e.g. `GlweCiphertext32`)